services/validateAPI/utils.py

```diff
@@ -0,0 +1,283 @@
+        return {
+            "type": "metadata_missing_sample",
+            "message": stderr.strip(),
+            "sample_id": validate_err_msg.group('sample_id')
+        }
+
+    validate_err_msg2 = re.search(
+        r'Invalid sample ID "(?P<sample_id>[^"]+)"',
+        stderr)
+    if validate_err_msg2:
+        logger.info("Parsed error: invalid_sample_id | %s", validate_err_msg2.group("sample_id"))
+        return {
+            "type": "invalid_sample_id",
+            "message": stderr.strip(),
+            "sample_id": validate_err_msg2.group('sample_id')
+        }
+
+    validate_err_msg3 = re.search(
+        r'Invalid Taxa of Interest: "(?P<value>[^"]+)"',
+        stderr)
+    if validate_err_msg3:
+        value = validate_err_msg3.group("value")
+        logger.info("Parsed error: invalid_taxa_of_interest | %s", value)
+        return {
+            "type": "invalid_taxa_of_interest",
+            "value": value,
+            "message": (
+                f'Taxa of Interest "{value}" is invalid. '
+                "Please provide a valid taxonomic name or remove it."
+            )
+        }
+
+    validate_err_msg4 = re.search(
+        r'Invalid Preliminary Morphology ID taxon "(?P<value>[^"]+)"',
+        stderr
+    )
+    if validate_err_msg4:
+        value = validate_err_msg4.group("value")
+        logger.info("Parsed error: invalid_pmi | %s", value)
+        return {
+            "type": "invalid_pmi",
+            "value": value,
+            "message": (
+                "The Preliminary Morphology ID is invalid,"
+                "Only letters (A–Z) and spaces are allowed. "
+                "Please fix this in the metadata CSV."
+            )
+        }
+
+    validate_err_msg5 = re.search(
+        r'The country provided could not be recognised: "(?P<value>[^"]+)"',
+        stderr
+    )
+    if validate_err_msg5:
+        value = validate_err_msg5.group("value")
+        logger.info("Parsed error: invalid_country | %s", value)
+        suggestions = {
+            "turkey": 'Use ISO alpha-2 code "TR".',
+            "türkiye": 'Use ISO alpha-2 code "TR".',
+            "hawaii": 'Hawaii is a US state. Please use "US".',
+        }
+        hint = suggestions.get(value.lower())
+        message = (
+            f'Country "{value}" is not recognised. '
+            + (hint if hint else
+                'Please replace it with a valid country name '
+                'or ISO alpha-2 code.')
+        )
+        return {
+            "type": "invalid_country",
+            "value": value,
+            "message": message
+        }
+
+    # Generic FASTA errors
+    if 'FASTAFormatError' in stderr:
+        logger.error("FASTA format error")
+        return {"type": "fasta_error", "message": stderr.strip()}
+
+    # Fallback
+    logger.error("Unknown validation error")
+    return {"type": "unknown", "message": stderr.strip()}
```

Comment:
Also here, returning an instance of a custom `ParsedError` class gives you a similar interface to the dict, but attributes are a bit nicer to work with than dict keys (your code editor will autocomplete them).